test/e2e/main_test.go

```diff
@@ -211,17 +212,19 @@ func testInstall(
 	// create a directory used to store the @testName container and resource logs
 	testOutputDir := filepath.Join(df.LoggingPath, testName)
 	err := os.Mkdir(testOutputDir, 0777)
-	assert.NoError(t, err, "creating the test case output directory should produce no error")
+	require.NoError(t, err, "creating the test case output directory should produce no error")
 
 	// randomize the namespace to avoid existing namespaces
 	rand.Seed(time.Now().UnixNano())
 	namespace := df.NamespacePrefix + "-" + strconv.Itoa(rand.Intn(50))
 
 	deployerCtx, err := df.NewDeployerCtx(repo, tag, namespace, testOutputDir, targetPods, spec)
-	assert.NoError(t, err, "creating a new deployer context should produce no error")
+	require.NoError(t, err, "creating a new deployer context should produce no error")
+	require.NotNil(t, deployerCtx, "the DeployerCtx object should not be nil")
 
 	rf, err := deployerCtx.Setup()
 	assert.NoError(t, err, "deploying metering should produce no error")
+	assert.NotNil(t, rf, "the reportingframework should not be nil")
```

Comment:
require?

Reply:
I used `require` here, then any statement that evaluates to false will end the test function, and the defer closure won't be run.